raster-foundry/raster-foundry
# Finish supporting Planet browsing

## Overview
* Handle repositories (Raster Foundry, Planet, more TBD) generically and refactor the browse view
* Allow creating imports for Planet scenes by creating imports directly from the scene browser. 
  * Some feedback / design work is needed to make this process more clear to the user.
* Consolidate several controller/component files
* Attempt an initial fix for dev environment logging out prematurely
* Minor api change for annotations actions (parameters were not in a consistent order)

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Notes
I've temporarily removed the grid layer since it's going to be replaced by the shape filters later anyways

## Testing Instructions

 * Verify that filters work as expected
* Verify that trying to switch to the planet repo when you don't have an api token is handled
* Verify that url parameters are reflected in the url, and loading / reloading the page reads filter parameters from the url
* Verify that annotations still work
* Verify that adding RF scenes to projects still works
* Verify that adding planet scenes creates a separate upload for each datasource
* Make sure I didn't make comments and forget to implement something

Closes #2887 
Closes #1478
